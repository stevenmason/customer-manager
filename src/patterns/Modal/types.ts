interface Props {
  onClose?: () => void;
}

export default Props;
