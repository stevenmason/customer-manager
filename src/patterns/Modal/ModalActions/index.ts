import ModalActions from "./ModalActions";
export default ModalActions;
