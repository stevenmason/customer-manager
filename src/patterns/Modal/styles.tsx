import styled from "styled-components";

const Container = styled.div`
  position: fixed;
  z-index: 1300;
  right: 0px;
  bottom: 0px;
  top: 0px;
  left: 0px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Modal = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 600px;
  background-color: #fff;
  border: 1px solid #ced4da;
  color: #5a6066;
  border-radius: 0.25rem;
  font-size: 1em;
  max-height: calc(100% - 96px);
  box-shadow: 0px 11px 15px -7px rgba(0, 0, 0, 0.2),
    0px 24px 38px 3px rgba(0, 0, 0, 0.14), 0px 9px 46px 8px rgba(0, 0, 0, 0.12);
`;

const Backdrop = styled.div`
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.5);
`;

export { Container, Backdrop, Modal };
