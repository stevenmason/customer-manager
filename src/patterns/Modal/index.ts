import Modal from "./Modal";
import ModalBody from "./ModalBody";
import ModalTitle from "./ModalTitle";
import ModalActions from "./ModalActions";

export default Modal;
export { ModalBody, ModalTitle, ModalActions };
