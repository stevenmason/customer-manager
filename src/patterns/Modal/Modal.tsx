import React from "react";

import Props from "./types";
import * as SC from "./styles";

const Modal: React.FC<Props> = ({ children, onClose }) => (
  <SC.Container>
    <SC.Backdrop onClick={onClose} />
    <SC.Modal>{children}</SC.Modal>
  </SC.Container>
);

export default Modal;
