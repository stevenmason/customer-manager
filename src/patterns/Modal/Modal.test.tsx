import React from "react";
import renderer from "react-test-renderer";

import Modal from "./Modal";

const PROPS = { onClose: () => {} };

it("renders correctly", () => {
  const tree = renderer.create(<Modal {...PROPS} />).toJSON();
  expect(tree).toMatchSnapshot();
});
