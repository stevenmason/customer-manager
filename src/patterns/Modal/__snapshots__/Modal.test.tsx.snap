// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa eksIDc"
>
  <div
    className="sc-htpNat jabHwM"
    onClick={[Function]}
  />
  <div
    className="sc-bwzfXH hXDBTd"
  />
</div>
`;
